samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,224 @@
+public class FileTransferSampleTest {
+
+    static final ObjectMapper MAPPER = new ObjectMapper();
+    static final String INITIATE_CONTRACT_NEGOTIATION_URI = "http://localhost:9192/api/v1/data/contractnegotiations";
+    static final String LOOK_UP_CONTRACT_AGREEMENT_URI = "http://localhost:9192/api/v1/data/contractnegotiations/{id}";
+    static final String INITIATE_TRANSFER_PROCESS_URI = "http://localhost:9192/api/v1/data/transferprocess";
+    static final String CONTRACT_OFFER_FILE_PATH = "samples/04.0-file-transfer/contractoffer.json";
+    static final String CONSUMER_CONFIG_PROPERTIES_FILE_PATH = "samples/04.0-file-transfer/consumer/config.properties";
+    static final String PROVIDER_CONFIG_PROPERTIES_FILE_PATH = "samples/04.0-file-transfer/provider/config.properties";
+    static final String TRANSFER_FILE_PATH = "samples/04.0-file-transfer/filetransfer.json";
+    // Reuse an already existing file for the test. Could be set to any other existing file in the repository.
+    static final String SAMPLE_ASSET_FILE_PATH = TRANSFER_FILE_PATH;
+    static final String DESTINATION_FILE_PATH = "samples/04.0-file-transfer/consumer/requested.test.txt";
+    static final Duration TIMEOUT = Duration.ofSeconds(15);
+    static final Duration POLL_INTERVAL = Duration.ofMillis(1000);
+    static final String API_KEY_HEADER_KEY = "X-Api-Key";
+    static final String API_KEY_HEADER_VALUE = "password";
+    @RegisterExtension
+    static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            "provider",
+            Map.of(
+                    // Override 'edc.samples.04.asset.path' implicitly set via property 'edc.fs.config'.
+                    "edc.samples.04.asset.path", getFileFromRelativePath(SAMPLE_ASSET_FILE_PATH).getAbsolutePath(),
+                    "edc.fs.config", getFileFromRelativePath(PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
+            "consumer",
+            Map.of(
+                    "edc.fs.config", getFileFromRelativePath(CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    static final File DESTINATION_FILE = getFileFromRelativePath(FileTransferSampleTest.DESTINATION_FILE_PATH);
+    String contractNegotiationId;
+    String contractAgreementId;
+
+    /**
+     * Resolves a {@link File} instance from a relative path.
+     */
+    @NotNull
+    static File getFileFromRelativePath(String relativePath) {
+        return new File(TestUtils.findBuildRoot(), relativePath);
+    }
+
+    /**
+     * Run all sample steps in one single test.
+     * Note: Sample steps cannot be separated into single tests because {@link EdcRuntimeExtension}
+     * runs before each single test.
+     */
+    @Test
+    void runSampleSteps() throws IOException {
```

Comment:
```suggestion
    void runSampleSteps() throws Exception {
```
nit: Is is easier for maintaining *test* code to throw general Exception class than to have to maintain in multiple places